### ISSUE TITLE
COS bucket: archive_rule overrides currently set up expiration rule

### ISSUE THREAD

**Issue by serg-lavr** [IBM-Cloud/terraform-provider-ibm#2068]
After update to a new version of ibm-cloud-terraform provider all currently set up expiration policies are deleted during terraform apply. 

Example output of terraform plan:

 ```
 Terraform will perform the following actions:

  # ibm_cos_bucket.bucket will be updated in-place
~ resource "ibm_cos_bucket" "bucket" {
        bucket_name          = "example-bucket-name"
        crn                  = "crn:v1:bluemix:public:cloud-object-storage:global:a/c7d87d30939146669d3f2bb43b2f1be7:3544f925-b0f3-4dc4-8771-a2ca0c7082b9:bucket:example-bucket-name"
        id                   = "crn:v1:bluemix:public:cloud-object-storage:global:a/c7d87d30939146669d3f2bb43b2f1be7:3544f925-b0f3-4dc4-8771-a2ca0c7082b9:bucket:example-bucket-name:meta:rl:us-east"
        region_location      = "us-east"
        resource_instance_id = "crn:v1:bluemix:public:cloud-object-storage:global:a/c7d87d30939146669d3f2bb43b2f1be7:3544f925-b0f3-4dc4-8771-a2ca0c7082b9::"
        s3_endpoint_private  = "s3.private.us-east.cloud-object-storage.appdomain.cloud"
        s3_endpoint_public   = "s3.us-east.cloud-object-storage.appdomain.cloud"
        storage_class        = "standard"

    - archive_rule {
        - days    = 0 -> null
        - enable  = true -> null
        - rule_id = "remove-data-60d-old" -> null
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.
 ```

### Terraform Version
```
v0.12.29
IBM Cloud Provider version 1.15.0
```
### Affected Resource(s)
- ibm_cos_bucket

### Terraform Configuration Files
Example config:
```
resource ibm_cos_bucket bucket {
  bucket_name          = "example-bucket-name"
  resource_instance_id = "example_cos_resource_instance_id"
  region_location      = us-east
  storage_class        = "standard"
}
```
### Expected Behavior
Expiration rules should be left without changes.
### Actual Behavior
Expiration rules removed after terraform apply.
### Steps to Reproduce
1. Create a bucket in terraform
2. Set up expiration policies in console
3. Run terraform apply again

**Comment by hkantare** [comment]
Its working as per desgin since you created the expiration configuration outside Terraform its expecting a diff  because when we run terraform apply it first invokes a refresh and updates the state information..(archieve rule will be refreshed in state)

But in configuration tempalte we can't find the declaration so it assumes user is trying to remove archieve rule configuation...

Update the tempalte with archieve_rule declartion and then run terraform plan it doesn't show any diff

**Comment by serg-lavr** [comment]
Thank you for response, @hkantare! The problem is, according to [cos_bucket docs](https://registry.terraform.io/providers/IBM-Cloud/ibm/latest/docs/resources/cos_bucket) `archive_rule` is used for archive policies and can be only of type "Glacier" and "Accelerated". I can see no way to set expiration rule in archive rule.

**Comment by serg-lavr** [comment]
@hkantare Just to clarify, my aim is to delete file after set amount of time, not to transfer it to archive. If the policy to delete is set in `archive_rule`, could you provide me with instruction on how to do this?

**Comment by Priyaranimishra** [comment]
@serg-lavr We are looking into the code and working on it. As of now archive_rule support is available. Deletion of the files after specific amount of time(`Expiration rule`) is not yet configured via a terraform.

**Comment by wderezin** [comment]
@Priyaranimishra take a look at #2079 to resolve this to see if you like the solution.   I was working on adding COS expire that happens to resolve this issue as well.  It still needs test cases & testing but the resource code is in place.

**Comment by Priyaranimishra** [comment]
Thank you @wderezin for raising initial draft of expiration rule. It looks good.

**Comment by mschenk42** [comment]
To be clear here, the new archive rules are incorrectly implemented. The archive resource is using Expiration rule configuration. Is there a workaround for this problem?

<img width="918" alt="Screenshot of kitty (12-4-20, 2-12-06 PM)" src="https://user-images.githubusercontent.com/630550/101210201-b3f34600-363a-11eb-8b4c-cb982a77525d.png">

<img width="1209" alt="Screenshot of Firefox (12-4-20, 2-13-40 PM)" src="https://user-images.githubusercontent.com/630550/101210394-059bd080-363b-11eb-8ad7-b5658a95d7d8.png">

**Comment by hkantare** [comment]
COS team is already looking into it ..As a workaround can you please add life_cycle ingnore block
lifecycle {
    ignore_changes = [
      "archive_rule"
    ]
  }

**Comment by wderezin** [comment]
@mschenk42  The archive and expire both use the same LifeCycle Configuration XML.  So the UI needs to parse the config and only display the the appropriate configs in each section.  When doing PR #2079 for COS Expire I had to merge the archive and expire rules into a single lifecycle config.

**Comment by wderezin** [comment]
@mschenk42 The fix comes with #2079 when you can manage both the archive and expire rules from terraform.  It is an all or nothing terraform management approach because the LifeCycle is a single API request for both Archive and Expire.   I don't know of a workout to this problem unless API is updated to manage the rules separately.
